app/uk/gov/hmrc/gform/views/summary/TextFormatter.scala

```diff
@@ -18,40 +18,73 @@ package uk.gov.hmrc.gform.views.summary
 
 import cats.syntax.option._
 import play.api.i18n.Messages
-import uk.gov.hmrc.gform.commons.BigDecimalUtil.toBigDecimalSafe
+import uk.gov.hmrc.gform.commons.BigDecimalUtil._
 import uk.gov.hmrc.gform.eval.smartstring.SmartStringEvaluator
 import uk.gov.hmrc.gform.fileupload.Envelope
 import uk.gov.hmrc.gform.models.Atom
 import uk.gov.hmrc.gform.models.helpers.DateHelperFunctions
-import uk.gov.hmrc.gform.sharedmodel.LocalisedString
+import uk.gov.hmrc.gform.sharedmodel.{ LangADT, LocalisedString, SmartString }
 import uk.gov.hmrc.gform.sharedmodel.formtemplate._
 import uk.gov.hmrc.gform.validation.{ FormFieldValidationResult, HtmlFieldId }
 import uk.gov.hmrc.gform.commons.NumberFormatUtil._
-import uk.gov.hmrc.gform.sharedmodel.LangADT
 
 object TextFormatter {
 
-  def componentTextReadonly(currentValue: String, textConstraint: TextConstraint)(implicit l: LangADT): String =
-    textConstraint match {
-      case PositiveNumber(_, fracDigits, rm, unit) => formatNumber(currentValue, fracDigits, rm, unit)
-      case Number(_, fracDigits, rm, unit)         => formatNumber(currentValue, fracDigits, rm, unit)
-      case s: Sterling                             => formatSterling(currentValue)
-      case _                                       => currentValue
+  def componentTextReadonly(
+    currentValue: String,
+    textConstraint: TextConstraint,
+    prefix: Option[SmartString] = None,
+    suffix: Option[SmartString] = None)(
```

Comment:
Addition of these two params is weird. There should be rather special implementation for Summary page purposes. Also it produces really weird result for component such as 
```
{
  "id": "number2",
  "type": "text",
  "label": "Hidden Field",
  "submitMode": "readonly",
  "value": "${'123'}",
  "format": "number(10, 2, 'litr')",
  "suffix": "suf",
  "prefix": "pref"
}
```

Reply:
Fixed.